src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -673,11 +668,21 @@ await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAs
                 // If the key-value was found, store it for updating the settings
                 if (watchedKv != null)
                 {
-                    watchedSettings[watchedKeyLabel] = watchedKv;
+                    watchedSettings[watchedKeyLabel] = new ConfigurationSetting(watchedKv.Key, watchedKv.Value, watchedKv.Label, watchedKv.ETag);
                     existingSettings[watchedKey] = watchedKv;
                 }
             }
 
+            foreach (KeyValueWatcher changeWatcher in _options.MultiKeyWatchers)

```

Comment:
This doesn't seem to have to do with loading the key-values registered for refresh.

Reply:
I agree, I can rename the method to something like UpdateWatchedSettings instead

Reply:
I think the key vaule collection piece needs to be pulled out of this method.

Reply:
Oh I see, I can work on that then

Reply:
Moving the code into a private method doesn't really address the concern. The state modification is still taking place in the `LoadKeyValuesRegisteredForRefresh` method.

By pulled out of the method, I meant that it should not happen in the `LoadKeyValuesRegisteredForRefresh` method.

Reply:
Ok sorry about that, so instead does it make sense to just create this new private method and call it each time after LoadKeyValuesRegisteredForRefresh is called so it's not taking place in that method?

Reply:
That works. Although the name shouldn't be `Load...` because it's not loading anything.

Reply:
I almost feel like we need a wrapper around these three calls now. There are two places we need to call these methods in the same order. How about something like this:

```
LoadAndWatchConfigurationSettings()
{
    LoadSelectedKeyValues()
    LoadKeyValuesRegisteredForRefresh()
    UpdateWatchedKeyValueCollection()
}
```

Reply:
Amer pointed out that we need to return two different dictionaries from these methods, data and watchedSettings. So to have a wrapper around the three methods, we would need to either return both dictionaries as a tuple/valuetuple, or make `LoadAndWatchConfigurationSettings` take out parameters. I think it would be better to return two dictionaries.  @jimmyca15 thoughts?

Reply:
I would just skip having the wrapper. Returning multiple types seems more complicated than having 3 method calls in two places.